### PR TITLE
Fix stacktrace response when dependent CHS service unavailable

### PR DESCRIPTION
**Pull request by hepsimo**
Due to api-sdk-java handling of CHS service unavailable, it throws
```IllegalArgumentException("")```
with cause
```IllegalArgumentException("expected numeric type but got class uk.gov.companieshouse.api.error.ApiErrorResponse")```

that caused RestExceptionHandler to fail because of the empty exception message.

Fix to handle the exception gracefully and produce a structured error.
Includes logging the probable cause to aid support team diagnosing the problem.

### CODE REVIEW

File: src/test/java/uk/gov/companieshouse/pscfiling/api/error/RestExceptionHandlerTest.java [hunk 1]
```diff
@@ -447,21 +479,22 @@ void handleAllUncaughtException(final Exception cause) {
...
-                        cause)));
+        final var causeCause = new ArithmeticException("DIV/0");
+
+        return Stream.of(
+            Arguments.of(new PscServiceException("PSC details unavailable", causeCause)),
+            Arguments.of(
+                new TransactionServiceException("Transaction details unavailable", causeCause)),
+            Arguments.of(
+                new CompanyProfileServiceException("Company profile details unavailable", null)));
     }
```
Comment (eedwards0) [inline src/test/java/uk/gov/companieshouse/pscfiling/api/error/RestExceptionHandlerTest.java:498]: Is it worth adding a test that contains a null String as well?

**Comment by eedwards0** [comment]
All JUnit unit tests and integration tests are passing. Error handling behaviour is as expected when running in Docker.

**Review by eedwards0 — APPROVED** [review]
All tests passing and error response when the transaction service is down is now handled as expected.
